### ISSUE TITLE
Missing ipproto keyword in network config rule section

### ISSUE THREAD

**Issue by TheRainstorm** [openwrt/luci#7330]
The `ip-rule`[^1] utility supports the `ipproto` keyword, allowing us to route UDP traffic differently using the following command:

```shell
ip ru add from all ipproto udp table udp_table prior 10
```

The `ip` utility in Openwrt also supports this keyword. Howerver, In Luci, there is no option to configure the `ipproto` keyword. The network configuration file `/etc/configure/network` also does not support it either. [^2]

As a result, to implement routing according to L4 protocol, we can only use command line or route using `fwmark` with the help of nftables. It would be more convenient if support for the `ipproto` keyword were added.

## Steps to reproduce:

1. go to: Network → Routing→ IPv4 rules
2. no method to set `ipproto`

## Additional Information:
OpenWrt version information from system `/etc/openwrt_release`

```
DISTRIB_ID='OpenWrt'
DISTRIB_RELEASE='23.05.0'
DISTRIB_REVISION='r23497-6637af95aa'
DISTRIB_TARGET='x86/64'
DISTRIB_ARCH='x86_64'
DISTRIB_DESCRIPTION='OpenWrt 23.05.0 r23497-6637af95aa'
DISTRIB_TAINTS=''
```

I have also installed `ip-full`

[^1]: https://man7.org/linux/man-pages/man8/ip-rule.8.html
[^2]: https://openwrt.org/docs/guide-user/network/routing/ip_rules

**Comment by systemcrash** [comment]
This is not complex to add within luci, but luci depends on what exists in the openwrt base system. I sent a patch for netifd to the mailing list and opened a PR which will probably take about 17 years to get acknowledged.

See https://github.com/openwrt/netifd/pull/36

**Comment by systemcrash** [comment]
Step 1 complete - change committed to netifd. 
Step 2: update openwrt to use newer netifd.

**Comment by systemcrash** [comment]
Closed by a6e3ea42ac9bbf7ba84f4f515c2c2986b172d234
